client/my-sites/invites/controller.js

```diff
@@ -14,10 +14,24 @@ import titleActions from 'lib/screen-title/actions';
 import InviteAccept from 'my-sites/invites/invite-accept';
 import { setSection } from 'state/ui/actions';
 import { renderWithReduxStore } from 'lib/react-helpers';
-import { acceptedNotice, getRedirectAfterAccept } from 'my-sites/invites/utils';
-import { successNotice } from 'state/notices/actions';
+import { getRedirectAfterAccept } from 'my-sites/invites/utils';
+import { acceptInvite as acceptInviteAction } from 'lib/invites/actions';
 
 export function acceptInvite( context ) {
+	const acceptedInvite = store.get( 'invite_accepted' );
+	if ( acceptedInvite ) {
+		store.remove( 'invite_accepted' );
+		const acceptInviteCallback = error => {
+			if ( error ) {
+				page( window.location.href );
```

Comment:
For what it's worth, I think the biggest flow issue is due to the force matching email issue, and once we get in #2647, this should be much better. So, we can probably leave this as is.